### ISSUE TITLE
GIT INFORMATION NOT APPEARING IN PROMPT

### ISSUE THREAD

**Issue by vanniktech** [ohmyzsh/ohmyzsh#12267]
### Describe the bug

I've just ran `omz update` and now it does not seem to work correctly; but neither do I know how I would debug this

### Steps to reproduce

Update from the version that I had to the current and suddenly my git configuration is gone.

### Expected behavior

It'd keep my git configuration as outlined in the first screenshot.

### Screenshots and recordings

Before it used to display everything correctly:

<img width="198" alt="Screenshot 2024-03-07 at 16 26 45" src="https://github.com/ohmyzsh/ohmyzsh/assets/5759366/7f9e294c-b603-4440-a6b2-8641bda51955">

But now it is missing all the informations:

<img width="112" alt="Screenshot 2024-03-07 at 16 27 04" src="https://github.com/ohmyzsh/ohmyzsh/assets/5759366/8e3cd0e1-e434-4f7c-be51-b1ca584e97b7">

### OS / Linux distribution

macOS Sonoma 14.3.1

### Zsh version

5.9

### Terminal emulator

iTerm2

### If using WSL on Windows, which version of WSL

None

**Comment by dianligegege** [comment]
I had the same problem, I switched to the previous version first.

`git checkout <commit-hash>`

**Comment by tyrken** [comment]
See workaround in #12257

**Comment by vivekbisen** [comment]
Introduced by this PR https://github.com/ohmyzsh/ohmyzsh/pull/12257

**Comment by vivekbisen** [comment]
> See workaround in #12257

tl;dr: Add this to your `.zshrc`
```zsh
zstyle ':omz:alpha:lib:git' async-prompt no

**Comment by carlosala** [comment]
The issue was isolated and we're working on the fix. Sorry for the noise.

**Comment by carlosala** [comment]
The new feature has been disabled temporarely. Running `omz update` will fix the problem.

**Comment by mcornella** [comment]
We found the culprit and I committed the fix in 06753e8. For now, the async prompt for git is turned off by default. You can however test it out with
```sh
zstyle ':omz:alpha:lib:git' async-prompt yes
```
We're quite proud of this one, and although it needs some work, we think it will get rid of all the "slow git prompt" issues, as it will now be rendered asynchronously.

Here's a recording of a before/after enabling on a shallow copy of the git repo https://github.com/torvalds/linux (which is a good test for slowness).

[![asciicast](https://asciinema.org/a/OBK9cA3Yc6kvU87BHm2jooL4v.svg)](https://asciinema.org/a/OBK9cA3Yc6kvU87BHm2jooL4v)

Please test and send in your feedback! Thanks!!

**Comment by vanniktech** [comment]
@mcornella that seems to be working fine on the first few runs. I must say that already on my small git repos I can feel the difference. Amazing work!

**Comment by mcornella** [comment]
This is now enabled by default since [ec1afe9](https://github.com/ohmyzsh/ohmyzsh/commit/ec1afe9dd683c36e6384db25fc1e95acbb0cbc7a) after fixing some logic and performance issues.

[Disabling it is done the same way](https://github.com/ohmyzsh/ohmyzsh#disable-async-git-prompt), but you should not experience any issues and instead get the full awesomeness of async prompt.

Please comment here or reopen if you have any issues!

**Comment by hwalinga** [comment]
I still face problems. How would I be able to debug this one?

**Comment by mcornella** [comment]
What problems? Ideally a screenshot or terminal output would be helpful.

**Comment by hwalinga** [comment]
Same problems as OP:

Before:

![image](https://github.com/ohmyzsh/ohmyzsh/assets/13176356/0092068d-c639-45bd-bbea-3fbffcbabef9)

After:

![image](https://github.com/ohmyzsh/ohmyzsh/assets/13176356/8a8f5ec2-5f54-467f-a7a7-c1cd9a06cd03)

theme: https://github.com/sobolevn/sobole-zsh-theme
zsh version: 5.9 (x86_64-debian-linux-gnu)
git version: 2.39.2
Debian GNU/Linux 12

**Comment by carlosala** [comment]
I see, your theme uses a [custom function](https://github.com/sobolevn/sobole-zsh-theme/blob/40ef40c1fb5eb5c1e7bbab913a7dbfbf1f449aa9/sobole.zsh-theme#L49-L51) print git information. We didn't take into account this situation. It's reproducible as well with `minimal` theme bundled in `oh-my-zsh`.

**Comment by coradinibr** [comment]
Same bug happening with theme pygmalion.
`zstyle ':omz:alpha:lib:git' async-prompt yes` or `no` doesn't make any difference.

**OS / Linux distribution**: macOS Sonoma 14.4.1
**Zsh version**: 5.9
**Terminal emulator**: iTerm2 3.4.23

**Comment by tomncooper** [comment]
@coradinibr I had the same issue, to fix it you have to switch the git info command in `$HOME/.oh-my-zsh/themes/pygmalion.zsh-theme`:
```
#local gitinfo=$(git_prompt_info)                                                       
local gitinfo=$(_omz_git_prompt_info)
```

**Comment by naveedkakal** [comment]
I was using a custom theme (bullet-train) and I needed to replace `git_prompt_info` call to `_omz_git_prompt_info` call and it worked again

**Comment by mcornella** [comment]
Don't do this:

> I was using a custom theme (bullet-train) and I needed to replace `git_prompt_info` call to `_omz_git_prompt_info` call and it worked again

Instead, add
```sh
zstyle ':omz:alpha:lib:git' async-prompt no
```
to your `.zshrc` file, before Oh My Zsh is sourced. This will temporarily disable the async prompt until we have a better fix. Relying on the current function names will break in the very near future.
